setup.sh

```diff
@@ -1,39 +1,54 @@
-ANDROID_NDK_PATH=/usr/local/ndk/25.1.8937393
+ANDROID_NDK_PATH="${ANDROID_NDK_PATH:-/usr/local/ndk/27.1.12297006}"
 if [[ ! `cat "${ANDROID_NDK_PATH}/CHANGELOG.md" 2> /dev/null` ]]; then
-    echo "no ndk found under /usr/local/ndk/25.1.8937393"
-    echo "download ndk 25.1.8937393 and create a symlink in '/usr/local/ndk/25.1.8937393' pointing to it"
+    log "no ndk found under ANDROID_NDK_PATH=${ANDROID_NDK_PATH}"
+    log "download ndk 27.1.12297006 and create a symlink in '/usr/local/ndk/27.1.12297006' pointing to it"
     exit 1
 fi
 
 if [[ ! ${ANDROID_ABI} ]]
 then
-    echo "ANDROID_ABI not set. Defaulting to 'arm64-v8a'"
+    log "ANDROID_ABI not set. Defaulting to 'arm64-v8a'"
     ANDROID_ABI=arm64-v8a
 fi
 
+: "${SCRIPT_ROOT:=$(cd "$(dirname "$0")"; pwd -P)}"
+
 readonly SDK_DIR="${SCRIPT_ROOT}/sdk"
-readonly TOOLCHAIN_PATH="${TOOLCHAIN_PATH:-/Library/Developer/Toolchains/swift-5.7-RELEASE.xctoolchain}"
+readonly TOOLCHAIN_PATH="${TOOLCHAIN_PATH:-/Library/Developer/Toolchains/swift-6.0.1-RELEASE.xctoolchain}"
 
 for LAST_ARGUMENT in $@; do :; done
 readonly PROJECT_DIRECTORY=${LAST_ARGUMENT:-$PWD}
 readonly BUILD_DIR="${PROJECT_DIRECTORY}/build/${ANDROID_ABI}"
 readonly LIBRARY_OUTPUT_DIRECTORY="${LIBRARY_OUTPUT_DIRECTORY:-${PROJECT_DIRECTORY}/libs/${ANDROID_ABI}}"
 readonly CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE:-"Debug"}
 
-readonly SWIFT_SDK_PATH="${SCRIPT_ROOT}/sdk/${ANDROID_ABI}"
+readonly SWIFT_SDK_PATH="${SCRIPT_ROOT}/sdk/swift-6.0.1-RELEASE-android-24-0.1.artifactbundle/swift-6.0.1-release-android-24-sdk/android-27b-sysroot"
+
 readonly HOST=darwin-x86_64 # TODO: add more platforms
 
 copySwiftDependencyLibs() {
+    log "Copying Swift dependency libraries..."
+
     function copyLib {
         local DESTINATION="${LIBRARY_OUTPUT_DIRECTORY}/`basename "$1"`"
+        local LIB_NAME=`basename "$1"`
+        log "Copying ${LIB_NAME}"
```

Comment:
[swift-android-toolchain] Copying Swift dependency libraries...
[swift-android-toolchain] Copying libGLESv2.so
[swift-android-toolchain] Copying libswiftAndroid.so
[swift-android-toolchain] Copying libandroid-spawn.so
[swift-android-toolchain] Copying libcamera2ndk.so
[swift-android-toolchain] Copying libmediandk.so
[swift-android-toolchain] Copying libstdc++.so
[swift-android-toolchain] Copying libssl.so
[swift-android-toolchain] Copying libOpenSLES.so
[swift-android-toolchain] Copying libGLESv3.so
[swift-android-toolchain] Copying libdispatch.so
[swift-android-toolchain] Copying libswiftRegexBuilder.so
[swift-android-toolchain] Copying libswift_Concurrency.so
[swift-android-toolchain] Copying libcrypto.so
[swift-android-toolchain] Copying libBlocksRuntime.so
[swift-android-toolchain] Copying libnghttp2.so
[swift-android-toolchain] Copying libssh2.so
[swift-android-toolchain] Copying libswiftCore.so
[swift-android-toolchain] Copying libnghttp3.so
[swift-android-toolchain] Copying libswiftSynchronization.so
[swift-android-toolchain] Copying libswift_StringProcessing.so
[swift-android-toolchain] Copying libjnigraphics.so
[swift-android-toolchain] Copying libiconv.so
[swift-android-toolchain] Copying libvulkan.so
[swift-android-toolchain] Copying libcharset.so
[swift-android-toolchain] Copying libm.so
[swift-android-toolchain] Copying libandroid.so
[swift-android-toolchain] Copying libc++_shared.so
[swift-android-toolchain] Copying libcares.so
[swift-android-toolchain] Copying libswiftDispatch.so
[swift-android-toolchain] Copying libswift_Builtin_float.so
[swift-android-toolchain] Copying liblog.so
[swift-android-toolchain] Copying liblzma.so
[swift-android-toolchain] Copying libz.so
[swift-android-toolchain] Copying libEGL.so
[swift-android-toolchain] Copying libGLESv1_CM.so
[swift-android-toolchain] Copying libOpenMAXAL.so
[swift-android-toolchain] Copying libz.so
[swift-android-toolchain] Copying libswift_RegexParser.so
[swift-android-toolchain] Copying libswift_math.so
[swift-android-toolchain] Copying libswiftObservation.so
[swift-android-toolchain] Copying libc.so
[swift-android-toolchain] Copying libdl.so
[swift-android-toolchain] Copying libc++.so
[swift-android-toolchain] Copying libc++_shared.so

Reply:
Hi @rikner that looks pretty good – I'm assuming it's after we already listed a number of EXCLUDED_LIBS in build.gradle?

To me it still looks at least half of them are not needed though. e.g. `libswiftObservation`, `libOpenMAXAL` `libswift_RegexParser` `libvulkan` `libjnigraphics` `libssh2` `nghttp3` `mediandk` `camera2ndk` `android-spawn` and probably more

Reply:
If we print it would also make sense to print the full path I think and not just the filename